webapp/src/pages/Settings.tsx

```diff
@@ -362,6 +381,67 @@ const Settings: React.FC<Props> = ({ open, onClose }) => {
       ),
     });
 
+  const displayPipelineAddSection = (pipelineIndex: number) => (
+    <Divider
+      sx={{
+        "&::before": { width: (theme) => theme.spacing(2) },
+        [`& .${dividerClasses.wrapper}`]: { padding: 0 },
+      }}
+    >
+      <Tooltip title="Add pipeline">
+        <IconButton
+          sx={{ color: (theme) => theme.palette.grey[400] }}
+          aria-label="add-pipeline"
+          disabled={isUpdatingConfig}
+          onClick={() =>
+            updatePartialConfig({
+              pipelines: [
+                ...resultingConfig.pipelines!.slice(0, pipelineIndex + 1),
+                ...defaultConfig.pipelines!,
```

Comment:
This seems dangerous. If one day we change the `defaultConfig` to have two pipelines, the button with start adding pairs if pipelines instead of just one. I suggest:
```suggestion
                defaultConfig.pipelines![0],
```